lib/db/mysql.js

```diff
@@ -281,7 +281,16 @@ module.exports = function (log, error) {
     return this.readFirstResult(
       CHECK_PASSWORD,
       [uid, hash.verifyHash]
-    )
+    ).then(function(result) {
+      return P.resolve(result)
+    }, function(err) {
+      // If .readFirstResult() doesn't find anything, it returns a error.notFound()
+      // so we need to convert that to an error.incorrectPassword()
+      if ( err.code === error.notFound().code && err.errno === error.notFound().errno ) {
```

Comment:
Yes, it's a bit weird that the `error.js` module returns a constant in some of those functions (which I guess mirrors the `.wrap()` function there which _does_ take a parameter).

Yes, let's file a follow up bug for that error stuff. I'll make sure to point to this code here. I'll also adjust the above to use just `errno`.